### PR TITLE
ci: reduce number of cache shards

### PR DESCRIPTION
**Pull request by gaurav274**
- [x] Fix memory failures on CI when running reuse test cases. An issue with `yolo` surfaces when the system is running on low memory. Explicitly calling garbage collection helped fix this issue.

### CODE REVIEW

File: eva/utils/kv_cache.py [hunk 1]
```diff
@@ -26,10 +26,10 @@ class DiskKVCache:
         `max_cache_size` (int, optional): an integer representing the maximum size of
             the cache. The system will make its best effort to enforce this limit, but it may slightly exceed it. The default value is 2**30.
         `shards` (int, optional): an integer representing the number of shards to use.
-            This can improve concurrent writes. The default value is 6.
+            This can improve concurrent writes. The default value is 3.
     """
 
-    def __init__(self, path: str, max_cache_size: int = 2**30, shards: int = 6):
+    def __init__(self, path: str, max_cache_size: int = 2**30, shards: int = 3):
```
Comment (jarulraj) [inline eva/utils/kv_cache.py:32]: This means 3 GB -- can we add a comment in the code itself?

File: eva/catalog/models/base_model.py [hunk 2]
```diff
@@ -108,7 +108,7 @@ def init_db():
         logger.info("Database does not exist, creating database.")
         create_database(engine.url)
         logger.info("Creating tables")
-        BaseModel.metadata.create_all()
+        BaseModel.metadata.create_all(checkfirst=True)
```
Comment (jarulraj) [inline eva/catalog/models/base_model.py:111]: What does this `checkfirst` option mean? Why is it now needed? We could add some comments here.

File: test/integration_tests/test_reuse.py [hunk 3]
```diff
@@ -44,6 +45,7 @@ def tearDown(self):
         execute_query_fetch_all("DROP TABLE IF EXISTS DETRAC;")
 
     def _verify_reuse_correctness(self, query, reuse_batch):
+        gc.collect()
```
Comment (jarulraj) [inline test/integration_tests/test_reuse.py:48]: Should we add a comment here for why this is needed? Links to relevant pages would also be helpful.

**Review by jarulraj — CHANGES_REQUESTED** [review]

File: eva/catalog/models/base_model.py (hunk 2 above)
Comment (gaurav274) [reply eva/catalog/models/base_model.py:111]: Removed. It is true by default.

File: eva/utils/kv_cache.py (hunk 1 above)
Comment (gaurav274) [reply eva/utils/kv_cache.py:32]: Done

**Pull request merged**